src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -1813,68 +1812,16 @@ protected boolean executeProgram(int dataSetNumber) {
-                    
-                } else {
-                    
-                    //OS supported (other values of osName could be "Linux", "SunOS", "FreeBSD", and "Mac OS X", all of which should work)
-                    //check if we're supposed to use the PC2 internal sandbox
-                    SandboxType sbType = problem.getSandboxType();
-                    if (sbType == SandboxType.PC2_INTERNAL_SANDBOX) {
-                        
-                        //copy the PC2 internal sandbox into the execution directory
-                        boolean success = generatePC2Sandbox();
-                        
-                        if (!success) {
-                            
-                            log.severe("Unable to generate PC2 Internal Sandbox; cannot execute submission");
-                            stderrlog.close();
-                            stdoutlog.close();
-                            executionData.setExecuteSucess(false);
-                            return false;
-                            
-                        }
-                        
-                    } else if (sbType == SandboxType.EXTERNAL_SANDBOX){
-
-                        //TODO: replace this block with whatever code is necessary to properly set up the specified external sandbox
-                        log.severe("Unsupported sandbox type '" + sbType +"' in Problem configuration; cannot execute submission");
-                        stderrlog.close();
-                        stdoutlog.close();
-                        executionData.setExecuteSucess(false);
-                        return false;
-                        
-                    } else {
-                        
-                        //unknown sandbox type
-                        log.severe("Unknown sandbox type '" + sbType +"' in Problem configuration; cannot execute submission");
-                        stderrlog.close();
-                        stdoutlog.close();
-                        executionData.setExecuteSucess(false);
-                        return false;
-                    }
-                }
-                
-                log.info("Using sandbox type '" + problem.getSandboxType() + "'; problem memory limit = " + problem.getMemoryLimitMB() + "MB");
-                
-                //Problem has a sandbox and we're not running a Team client; ensure we run the sandbox
-                usingSandbox = true;
+            //determine whether the Problem is configured to use a sandbox 
+            try {
+                usingSandbox = isUsingSandbox();
+            } catch (Exception e) {
+                //an exception means there is something wrong about sandbox usage (e.g. not allowed on this platform)
+                log.severe("Exception during sandbox usage check: " + e.getMessage());
+                stderrlog.close();
+                stdoutlog.close();
+                executionData.setExecuteSucess(false);
```

Comment:
Please add the following line before the return
executionData.setExecutionException(e);

Until we refeactor executable to throw exceptions, it is helpful to save the exception so we
can access it later  using the executionData.   Adding the above line gives us that access.

Reply:
Good point; done.